src/main/java/com/rapid7/client/dcerpc/mslsad/objects/SAMPRDomainPasswordInfo.java

```diff
@@ -0,0 +1,160 @@
+     * following table. All bits can be combined using a logical OR in any combination. Undefined bits SHOULD be
+     * persisted by the server (that is, stored in its database) and returned to future queries. Clients SHOULD ignore
+     * undefined bits.
+     *
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | Name/value                      | Description                                                              |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_PASSWORD_COMPLEX         | The server enforces password complexity policy.                          |
+     * | 0x00000001                      |                                                                          |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_PASSWORD_NO_ANON_CHANGE  | Reserved. No effect on password policy.                                  |
+     * | 0x00000002                      |                                                                          |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_PASSWORD_NO_CLEAR_CHANGE | Change-password methods that provide the cleartext password are disabled |
+     * | 0x00000004                      | by the server.                                                           |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_LOCKOUT_ADMINS           | Reserved. No effect on password policy.                                  |
+     * | 0x00000008                      |                                                                          |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_PASSWORD_STORE_CLEARTEXT | The server MUST store the cleartext password, not just the computed      |
+     * | 0x00000010                      | hashes.                                                                  |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_REFUSE_PASSWORD_CHANGE   | Reserved. No effect on password policy.                                  |
+     * | 0x00000020                      |                                                                          |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     */
+    private short minimumPasswordLength;
+    private short passwordHistoryLength;
+    private int passwordProperties;
+    private int maximumPasswordAge;
```

Comment:
Both `maximumPasswordAge` and `minimumPasswordAge` should be `long`'s. The documentation states that they are `LARGE_INTEGER` unions, which convert directly to Java's signed long.
```
typedef struct _OLD_LARGE_INTEGER {
  ULONG LowPart;
  LONG  HighPart;
} OLD_LARGE_INTEGER, *POLD_LARGE_INTEGER;

LowPart
   Low-order 32 bits.
HighPart
   High-order 32 bits. The sign of this member determines the sign of the 64-bit integer.
```

Now, if we were using big endian, we would have a problem here, because your MSB->LSB would look like this on the wire:
`4 3 2 1 4 3 2 1`
When calling `readLong()`, you would interpret this incorrectly, and so would require special handling here.

However, since everything in dcerpc is forced to be little endian, we can take a shortcut here and simply call `readLong()` to retrieve the value. This is because your MSB->LSB ordering would be this:
`1 2 3 4 1 2 3 4`
which when we consider the LowPart prefixes the HighPart, would read exactly as expected when calling readLong()`:
`1 2 3 4 5 6 7 8`